File: frontend/src/types/task.ts
export interface Task {
  id: number
  title: string
  description?: string
  completed: boolean
  createdAt: string
  expireAt?: string
  updatedAt: string
  version: number
}

export interface TaskRequest {
  title: string
  description?: string
  completed?: boolean
  expireAt?: string
  version?: number
}

File: backend/src/main/java/com/delta/taskmanagement/entity/Task.java
//[AI_assisted_001]
package com.delta.taskmanagement.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

@Entity
@Table(name = "tasks")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    @Builder.Default
    private boolean completed = false;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;

    @Column(nullable = true)
    private Instant expireAt;

    @Version
    private Long version;
}
//[AI_assisted_001]